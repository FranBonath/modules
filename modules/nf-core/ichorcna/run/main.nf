process ICHORCNA_RUN {
    tag "$meta.id"
    label 'process_low'

    conda "bioconda::r-ichorcna=0.5.0"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/r-ichorcna:0.5.0--pl5321r42hdfd78af_0' :
        'biocontainers/r-ichorcna:0.5.0--pl5321r42hdfd78af_0' }"

    input:
    tuple val(meta), path(wig)
    path gc_wig
    path map_wig
    path normal_wig
    path normal_background
    path centromere
    path rep_time_wig
    path exons

    output:
    tuple val(meta), path("**.cna.seg")    , emit: cna_seg
    tuple val(meta), path("**.params.txt") , emit: ichorcna_params
    tuple val(meta), path("**.pdf")        , emit: genome_plot
    path "versions.yml"                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args       ?: ''
    def prefix = task.ext.prefix   ?: "${meta.id}"
    def norm   = normal_wig        ? "normal_wig='${normal_wig}',"          : 'normal_wig=NULL,'
    def pon    = normal_background ? "normal_panel='${normal_background}'," : 'normal_panel=NULL,'
    def map    = map_wig           ? "mapWig='${map_wig}',"                 : 'mapWig=NULL,'
    def centro = centromere        ? "centromere='${centromere}',"          : ''
    def rep    = rep_time_wig      ? "repTimeWig='${rep_time_wig}',"        : 'repTimeWig=NULL,'
    def exons  = exons             ? "exons.bed='${exons}',"                : ''
    """
    #!/usr/bin/env Rscript
    library("ichorCNA")
    library("yaml")

    run_ichorCNA(
        tumor_wig='${wig}',
        id='${prefix}',
        cores=${task.cpus},
        gcWig='${gc_wig}',
        $norm
        $pon
        $map
        $centro
        $rep
        $exons
        $args
        outDir="."
    )


    ### Make Versions YAML for NF-Core ###
    versions = list()
    versions["r"]        <- paste(R.Version()\$major, R.Version()\$minor, sep=".")
    versions["ichorCNA"] <- paste(packageVersion("ichorCNA"), sep=".")

    yaml_str <- as.yaml(
        list(
            "${task.process}" = versions
        )
    )
    writeLines(yaml_str, file("versions.yml"))
    """
}
