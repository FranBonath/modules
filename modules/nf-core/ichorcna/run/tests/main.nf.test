nextflow_process {

    name "Test Process ICHORCNA_RUN"
    script "../main.nf"
    process "ICHORCNA_RUN"

    tag "modules"
    tag "modules_nfcore"
    tag "ichorcna"
    tag "ichorcna/run"

    test("no_panel") {
        config "./nextflow.config"
        when {
            process {
                """
                input[0] = [ [ id:'test'], // meta map
                       file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/MBC_315.ctDNA.reads.wig", checkIfExists: true)
                     ]
                input[1] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/gc_hg19_1000kb.wig", checkIfExists: true)
                input[2] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/map_hg19_1000kb.wig", checkIfExists: true)
                input[3] = []
                input[4] = []
                input[5] = file("https://raw.githubusercontent.com/gavinhalab/ichorCNA/master/inst/extdata/GRCh37.p13_centromere_UCSC-gapTable.txt", checkIfExists: true)
                input[6] = []
                input[7] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
