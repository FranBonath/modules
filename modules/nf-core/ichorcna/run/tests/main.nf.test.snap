{
    "nopanel - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "cna_seg": [
                    
                ],
                "genome_plot": [
                    
                ],
                "ichorcna_params": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-01T11:44:28.234644214"
    },
    "no_panel": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cna.seg:md5,a3e0998787901a129a582191a2459168"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.params.txt:md5,fa11ceb071512b7dd768003cc1d90647"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_CNA_chr1.pdf:md5,bdc29d2628fa145d4ab201ab1edb5862",
                            "test_CNA_chr10.pdf:md5,82e22e531a7ebb86349986bac3455423",
                            "test_CNA_chr11.pdf:md5,9f951a57247d30cda26791f49e68d1a8",
                            "test_CNA_chr12.pdf:md5,867394f0106ef761c252b237d540ca4b",
                            "test_CNA_chr13.pdf:md5,fdfb389ead30741c960d515f7add89ca",
                            "test_CNA_chr14.pdf:md5,3ec4ec6cc79e88b46fd4827808698129",
                            "test_CNA_chr15.pdf:md5,00ae85669aef3d6e90a4217e3f1a4b53",
                            "test_CNA_chr16.pdf:md5,da4ae36cd395f04b4a7f1c3c5e43407f",
                            "test_CNA_chr17.pdf:md5,a1034ee74379efddb515fad0d2afe9d1",
                            "test_CNA_chr18.pdf:md5,4407fd2b7028eb238d8c19a0cbbeeca2",
                            "test_CNA_chr19.pdf:md5,fd93c088ccba5041326c3f3db4fbe5f3",
                            "test_CNA_chr2.pdf:md5,6de164016ee0d31508c9f8f27a0368d6",
                            "test_CNA_chr20.pdf:md5,4c745d25e2fbf3a47588ecd01b92f7a3",
                            "test_CNA_chr21.pdf:md5,3b8632203c9ebdde8ec06156d25f6b95",
                            "test_CNA_chr22.pdf:md5,0f5a9078a32673222f1f931a9fc8c6db",
                            "test_CNA_chr3.pdf:md5,365221e1ce6be5867d5e3233956a655a",
                            "test_CNA_chr4.pdf:md5,9ebaab52e8095819522f5f005abfd64e",
                            "test_CNA_chr5.pdf:md5,535c1f5d9942abfa2c0555235be08797",
                            "test_CNA_chr6.pdf:md5,5efd07ef685172deb9ef0e85671ce670",
                            "test_CNA_chr7.pdf:md5,8ee047ac32f8e2912de796474f88fe77",
                            "test_CNA_chr8.pdf:md5,4ac317c7b6844cf63638a1cb6337e33b",
                            "test_CNA_chr9.pdf:md5,20a0779860771840d6f03e1cff656496",
                            "test_CNA_chrX.pdf:md5,91a58fd6d3c6c3b39a59f5ed1f1fcd94",
                            "test_bias.pdf:md5,eca21b685112cf747bd245ced41f8ae5",
                            "test_correction_chr1.pdf:md5,d88d7ece6434d69a0520f648c82709d5",
                            "test_correction_chr10.pdf:md5,9e06af5027b1cd0f50aee12e20883e36",
                            "test_correction_chr11.pdf:md5,f60d5bf962f65a9c3c0dfcff4aeedf4e",
                            "test_correction_chr12.pdf:md5,1cfd53bed082984801a5a6dae1f16942",
                            "test_correction_chr13.pdf:md5,c5f6e73b2252eb451edcfd4ed0b4e968",
                            "test_correction_chr14.pdf:md5,0b7c204e10c1964b834f394b46be85bb",
                            "test_correction_chr15.pdf:md5,8a311de47748c6cc888a175fda9b1c2e",
                            "test_correction_chr16.pdf:md5,17530f2384cf8bdc980cf8fb554c910c",
                            "test_correction_chr17.pdf:md5,7e5ca5633362b3121a746709b392f69e",
                            "test_correction_chr18.pdf:md5,2c302dc3148d3f5a8221bdbb80930843",
                            "test_correction_chr19.pdf:md5,8e94b67020a15c431eac146547e83b19",
                            "test_correction_chr2.pdf:md5,0020ff8409c5aec170f9d89b0a4ed342",
                            "test_correction_chr20.pdf:md5,b072d7792344e7f22e8c0f736c5b7228",
                            "test_correction_chr21.pdf:md5,9c86b4bd45615c192a264045fbb0fb4f",
                            "test_correction_chr22.pdf:md5,d21a5fe9e3f40c62bca4b5aeabc2a0da",
                            "test_correction_chr3.pdf:md5,e1d2455aa5d1a9e24fefe80e1fe822f1",
                            "test_correction_chr4.pdf:md5,130acb533dae4df548d5e7c09804ef42",
                            "test_correction_chr5.pdf:md5,c0d74ead303d9d6f19e55502a6a85877",
                            "test_correction_chr6.pdf:md5,d8bd24bf35ab17874c9b2d5907f7cbd0",
                            "test_correction_chr7.pdf:md5,129c95ff894e27a8cab1022bd4e1ed41",
                            "test_correction_chr8.pdf:md5,260adda8b25ccfb4f153272644fbec88",
                            "test_correction_chr9.pdf:md5,dffce51feebdb689560e67d163a0eb5e",
                            "test_correction_chrX.pdf:md5,97a05c719180a373d74229dc3eba8f78",
                            "test_genomeWide.pdf:md5,8ab853e9f6e7ab619a74ae687b27e5b8",
                            "test_genomeWideCorrection.pdf:md5,12a1d206554bd7817da425ade24fe2a9",
                            "test_genomeWide_all_sols.pdf:md5,2a32ec6c40eab3426dcc5b9da9b7fc45",
                            "test_genomeWide_n0.5-p2.pdf:md5,2a32ec6c40eab3426dcc5b9da9b7fc45",
                            "test_tpdf.pdf:md5,01f740d58878e00382c6ee4b731c8060"
                        ]
                    ]
                ],
                "3": [
                    "versions.yml:md5,d2d30cedd49c7bffdcb74313e8a074c4"
                ],
                "cna_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cna.seg:md5,a3e0998787901a129a582191a2459168"
                    ]
                ],
                "genome_plot": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_CNA_chr1.pdf:md5,bdc29d2628fa145d4ab201ab1edb5862",
                            "test_CNA_chr10.pdf:md5,82e22e531a7ebb86349986bac3455423",
                            "test_CNA_chr11.pdf:md5,9f951a57247d30cda26791f49e68d1a8",
                            "test_CNA_chr12.pdf:md5,867394f0106ef761c252b237d540ca4b",
                            "test_CNA_chr13.pdf:md5,fdfb389ead30741c960d515f7add89ca",
                            "test_CNA_chr14.pdf:md5,3ec4ec6cc79e88b46fd4827808698129",
                            "test_CNA_chr15.pdf:md5,00ae85669aef3d6e90a4217e3f1a4b53",
                            "test_CNA_chr16.pdf:md5,da4ae36cd395f04b4a7f1c3c5e43407f",
                            "test_CNA_chr17.pdf:md5,a1034ee74379efddb515fad0d2afe9d1",
                            "test_CNA_chr18.pdf:md5,4407fd2b7028eb238d8c19a0cbbeeca2",
                            "test_CNA_chr19.pdf:md5,fd93c088ccba5041326c3f3db4fbe5f3",
                            "test_CNA_chr2.pdf:md5,6de164016ee0d31508c9f8f27a0368d6",
                            "test_CNA_chr20.pdf:md5,4c745d25e2fbf3a47588ecd01b92f7a3",
                            "test_CNA_chr21.pdf:md5,3b8632203c9ebdde8ec06156d25f6b95",
                            "test_CNA_chr22.pdf:md5,0f5a9078a32673222f1f931a9fc8c6db",
                            "test_CNA_chr3.pdf:md5,365221e1ce6be5867d5e3233956a655a",
                            "test_CNA_chr4.pdf:md5,9ebaab52e8095819522f5f005abfd64e",
                            "test_CNA_chr5.pdf:md5,535c1f5d9942abfa2c0555235be08797",
                            "test_CNA_chr6.pdf:md5,5efd07ef685172deb9ef0e85671ce670",
                            "test_CNA_chr7.pdf:md5,8ee047ac32f8e2912de796474f88fe77",
                            "test_CNA_chr8.pdf:md5,4ac317c7b6844cf63638a1cb6337e33b",
                            "test_CNA_chr9.pdf:md5,20a0779860771840d6f03e1cff656496",
                            "test_CNA_chrX.pdf:md5,91a58fd6d3c6c3b39a59f5ed1f1fcd94",
                            "test_bias.pdf:md5,eca21b685112cf747bd245ced41f8ae5",
                            "test_correction_chr1.pdf:md5,d88d7ece6434d69a0520f648c82709d5",
                            "test_correction_chr10.pdf:md5,9e06af5027b1cd0f50aee12e20883e36",
                            "test_correction_chr11.pdf:md5,f60d5bf962f65a9c3c0dfcff4aeedf4e",
                            "test_correction_chr12.pdf:md5,1cfd53bed082984801a5a6dae1f16942",
                            "test_correction_chr13.pdf:md5,c5f6e73b2252eb451edcfd4ed0b4e968",
                            "test_correction_chr14.pdf:md5,0b7c204e10c1964b834f394b46be85bb",
                            "test_correction_chr15.pdf:md5,8a311de47748c6cc888a175fda9b1c2e",
                            "test_correction_chr16.pdf:md5,17530f2384cf8bdc980cf8fb554c910c",
                            "test_correction_chr17.pdf:md5,7e5ca5633362b3121a746709b392f69e",
                            "test_correction_chr18.pdf:md5,2c302dc3148d3f5a8221bdbb80930843",
                            "test_correction_chr19.pdf:md5,8e94b67020a15c431eac146547e83b19",
                            "test_correction_chr2.pdf:md5,0020ff8409c5aec170f9d89b0a4ed342",
                            "test_correction_chr20.pdf:md5,b072d7792344e7f22e8c0f736c5b7228",
                            "test_correction_chr21.pdf:md5,9c86b4bd45615c192a264045fbb0fb4f",
                            "test_correction_chr22.pdf:md5,d21a5fe9e3f40c62bca4b5aeabc2a0da",
                            "test_correction_chr3.pdf:md5,e1d2455aa5d1a9e24fefe80e1fe822f1",
                            "test_correction_chr4.pdf:md5,130acb533dae4df548d5e7c09804ef42",
                            "test_correction_chr5.pdf:md5,c0d74ead303d9d6f19e55502a6a85877",
                            "test_correction_chr6.pdf:md5,d8bd24bf35ab17874c9b2d5907f7cbd0",
                            "test_correction_chr7.pdf:md5,129c95ff894e27a8cab1022bd4e1ed41",
                            "test_correction_chr8.pdf:md5,260adda8b25ccfb4f153272644fbec88",
                            "test_correction_chr9.pdf:md5,dffce51feebdb689560e67d163a0eb5e",
                            "test_correction_chrX.pdf:md5,97a05c719180a373d74229dc3eba8f78",
                            "test_genomeWide.pdf:md5,8ab853e9f6e7ab619a74ae687b27e5b8",
                            "test_genomeWideCorrection.pdf:md5,12a1d206554bd7817da425ade24fe2a9",
                            "test_genomeWide_all_sols.pdf:md5,2a32ec6c40eab3426dcc5b9da9b7fc45",
                            "test_genomeWide_n0.5-p2.pdf:md5,2a32ec6c40eab3426dcc5b9da9b7fc45",
                            "test_tpdf.pdf:md5,01f740d58878e00382c6ee4b731c8060"
                        ]
                    ]
                ],
                "ichorcna_params": [
                    [
                        {
                            "id": "test"
                        },
                        "test.params.txt:md5,fa11ceb071512b7dd768003cc1d90647"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d2d30cedd49c7bffdcb74313e8a074c4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-01T13:00:59.064766191"
    }
}